### PR TITLE
What's new in Liberty Tools for IntelliJ IDEA 0.0.8 early release blog post

### PR DESCRIPTION
**Pull request by kathrynkodama**
Part of https://github.com/OpenLiberty/blogs/issues/2868

https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/2023/02/14/liberty-tools-intellij-early-release.html

If you have any ideas on how to make the YouTube video appear larger on the blog post, that would be great. 

image

image

image

image

### CODE REVIEW

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+With the latest early release of Liberty Tools for IntelliJ IDEA - iterate fast with Liberty dev mode, easily edit Liberty configuration files and code with assistance for MicroProfile and Jakarta EE APIs, all from within the editor!
+
+Try it out today with IntelliJ IDEA 2022.2+, installable via the link:https://plugins.jetbrains.com/plugin/14856-liberty-tools/[Jetbrains marketplace].
+
+Check out the video below to see all the Liberty Tools for IntelliJ IDEA features:
+
+video::2T-ALsTGRY0[youtube]
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
.Deep Dive of Liberty Tools for IntelliJ IDEA (early release)
video::2T-ALsTGRY0[youtube, width="560", height="315"]
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+// image::img/blog/FILE_NAME[IMAGE CAPTION ,width=70%,align="center"]
+// -------------------------
+// "[.img_border_light]" = This adds a faint grey border around the image to make its edges sharper. Use it around screenshots but not           
+// around diagrams. Then double check how it looks.
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+With the latest early release of Liberty Tools for IntelliJ IDEA - iterate fast with Liberty dev mode, easily edit Liberty configuration files and code with assistance for MicroProfile and Jakarta EE APIs, all from within the editor!
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
With the latest early release of Liberty Tools for IntelliJ IDEA, you can iterate fast with Liberty dev mode, code with assistance for MicroProfile and Jakarta EE APIs, and easily edit Liberty configuration files, all from within the editor.
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+layout: post
+title: "Jakarta EE, MicroProfile, and Liberty configuration editing support in IntelliJ IDEA with Liberty Tools (early release)"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/kathrynkodama
+author_github: https://github.com/kathrynkodama
+seo-title: Jakarta EE, MicroProfile, and Liberty configuration editing support in IntelliJ IDEA with Liberty Tools (early release) - OpenLiberty.io
+seo-description: The latest cloud-native Java support offered by the Liberty Tools for IntelliJ IDEA early release. Iterate fast with Liberty dev mode, code with assistance for MicroProfile and Jakarta EE APIs, and easily edit Liberty configuration files, all from within the IntelliJ IDEA editor.
+blog_description: "The latest cloud-native Java support offered by the Liberty Tools for IntelliJ IDEA early release. Iterate fast with Liberty dev mode, code with assistance for MicroProfile and Jakarta EE APIs, and easily edit Liberty configuration files, all from within the IntelliJ IDEA editor."
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
open-graph-image: https://openliberty.io/img/blog/LT-intellij-0.0.8-LCLS-server-xml-completion.png
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+open-graph-image-alt: Open Liberty Logo
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
open-graph-image-alt: Code assistance with Liberty Tools
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+Try it out today with IntelliJ IDEA 2022.2+, installable via the link:https://plugins.jetbrains.com/plugin/14856-liberty-tools/[Jetbrains marketplace].
+
+Check out the video below to see all the Liberty Tools for IntelliJ IDEA features:
+
+video::2T-ALsTGRY0[youtube]
+
+== What's New
+
+=== Type-ahead completion for Jakarta EE, MicroProfile and Liberty configuration files
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
=== Type-ahead completion for Jakarta EE, MicroProfile, and Liberty configuration files
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+In this early release, we have added all new type-ahead completion assistance for your Liberty configuration files and Jakarta EE and MicroProfile APIs.
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
In this early release, we add all new type-ahead completion assistance for your Liberty configuration files and Jakarta EE and MicroProfile APIs.
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+Liberty configuration assistance provides code completion in Liberty `server.xml`, `server.env`, and `bootstrap.properties` files. For example, the completion shown below offers possible Liberty features you can configure in your `server.xml` file's `featureManager` block.
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
Liberty configuration assistance provides code completion in Liberty `server.xml`, `server.env`, and `bootstrap.properties` files. For example, the completion in the following example offers possible Liberty features you can configure in your `server.xml` file's `featureManager` block.
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+[.img_border_light]
+image::img/blog/LT-intellij-0.0.8-LCLS-server-xml-completion.png[Liberty configuration assistance provides code completion in Liberty `server.xml`, `server.env`, and `bootstrap.properties` files.,width=70%,align="center"]
+
+Editing assistance provides code completion in configuration and application files for Jakarta EE and MicroProfile APIs. For example, the `rest_get` code completion shown below inserts a code snippet for a RESTful Web Services GET method.
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
Editing assistance provides code completion in configuration and application files for Jakarta EE and MicroProfile APIs. For example, the `rest_get` code completion in the following example inserts a code snippet for a RESTful Web Services `GET` method.
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+[.img_border_light]
+image::img/blog/LT-intellij-0.0.8-LSP4Jakarta-rest-completion.png[Editing assistance provides code completion in configuration and application files for Jakarta EE and MicroProfile APIs.,width=70%,align="center"]
+
+=== Navigate using the *Go to Action* menu
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
=== Navigate using the Go to Action menu
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+If you prefer to work entirely through keyboard shortcuts, all Liberty actions can now be run through the IntelliJ *Go to Action* menu. You can access the menu either by pressing `Shift` twice, then selecting the Actions tab, or by pressing `Ctrl` + `Shift` + `A` / `Cmd` + `Shift` + `A`.
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
If you prefer to work entirely through keyboard shortcuts, all Liberty actions can now be run through the IntelliJ *Go to Action* menu. You can access the menu either by pressing `Shift` twice, then selecting the Actions tab, or by pressing `Ctrl` + `Shift` + `A`  or  `Cmd` + `Shift` + `A`.
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+[.img_border_light]
+image::img/blog/LT-intellij-0.0.8-liberty-go-to-action-view.png[Run Liberty actions using the keyboard with the IntelliJ Go To Action menu.,width=70%,align="center"]
+
+=== Run/Debug configuration integration
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
=== Attach a debugger at startup
```

or similar, focus the heading on what the user can do w/ this info. Use imperative voice to be parallel with the other headings

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+Selecting the Liberty *Start...* action from the Liberty tool window or IntelliJ *Go To Action* menu, you can configure a new run configuration with Maven or Gradle parameters that you want passed to the `mvn liberty:dev`/`gradle libertyDev` commands. This integrates into the IntelliJ IDEA run/debug configuration view.
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
When you select the Liberty *Start...* action from the Liberty tool window or IntelliJ *Go To Action* menu, you can configure a new run configuration with Maven or Gradle parameters to pass to the `mvn liberty:dev` or `gradle libertyDev` commands. This configuration integrates into the IntelliJ IDEA run/debug configuration view.
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+[.img_border_light]
+image::img/blog/LT-intellij-0.0.8-liberty-start-with-config.png[Start dev mode with run configuration.,width=70%,align="center"]
+
+Selecting the newly created Liberty run configuration and then selecting the IntelliJ *Debug* icon, your Liberty server will start and attach a JVM debugger all in one step. The *Debug* action will determine an available port on your system and pass that port to dev mode. As soon as dev mode starts, you can start debugging your application through the built-in IntelliJ IDEA debug view.
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
When you select the newly created Liberty run configuration and then select the IntelliJ *Debug* icon, your Liberty server will start and attach a JVM debugger all in one step. The *Debug* action will determine an available port on your system and pass that port to dev mode. As soon as dev mode starts, you can start debugging your application through the built-in IntelliJ IDEA debug view.
```

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+
+[.img_border_light]
+image::img/blog/LT-intellij-0.0.8-liberty-debug-confirm-config.png[Debug dev mode with run configuration.,width=70%,align="center"]
+
+For a full list of newly added features and bug fixes check out the link:https://github.com/OpenLiberty/liberty-tools-intellij/releases/tag/0.0.8[0.0.8 release notes].
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
For a full list of newly added features and bug fixes, check out the link:https://github.com/OpenLiberty/liberty-tools-intellij/releases/tag/0.0.8[0.0.8 release notes].
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review

File: posts/2023-02-14-liberty-tools-intellij-early-release.adoc [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+With the latest early release of Liberty Tools for IntelliJ IDEA - iterate fast with Liberty dev mode, easily edit Liberty configuration files and code with assistance for MicroProfile and Jakarta EE APIs, all from within the editor!
+
+Try it out today with IntelliJ IDEA 2022.2+, installable via the link:https://plugins.jetbrains.com/plugin/14856-liberty-tools/[Jetbrains marketplace].
+
+Check out the video below to see all the Liberty Tools for IntelliJ IDEA features:
```
Comment (dmuelle) [inline posts/2023-02-14-liberty-tools-intellij-early-release.adoc]:
```suggestion
Check out the following video to see all the Liberty Tools for IntelliJ IDEA features:
```

**Comment by kathrynkodama** [comment]
@dmuelle incorporated your edits, ready for re-review: https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/2023/02/14/liberty-tools-intellij-early-release.html

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Pull request merged**